test/index.jl

```diff
@@ -529,4 +529,41 @@ end
     @test crossjoin(dfv, dfv, makeunique=true) == DataFrame(a=1, a_1=1)
 end
 
+@testset "AsTable constructor" begin
+    @test_throws ArgumentError AsTable(1.0)
+    @test_throws ArgumentError AsTable(true)
+    @test_throws ArgumentError AsTable(false)
+    @test_throws ArgumentError AsTable(Any[1, missing])
+    @test_throws ArgumentError AsTable([1, missing])
+    @test_throws ArgumentError AsTable([true, missing])
+    @test_throws ArgumentError AsTable(Any[true, missing])
+    @test_throws ArgumentError AsTable([:A, missing])
+    @test_throws ArgumentError AsTable(Any[:A, missing])
+    @test_throws ArgumentError AsTable(1.0:1.0)
+    @test_throws ArgumentError AsTable([1.0])
+    @test_throws ArgumentError AsTable(Any[1.0])
+    @test_throws ArgumentError AsTable([:a, "a", 1, true, 1.0])
+    @test_throws ArgumentError AsTable(:a => :b)
+
+    @test_nowarn AsTable(Union{Bool, Missing}[true, false])
```

Comment:
What kind of warning could this print?

Reply:
The name of the macro is misleading. What it does is more general:

> Test whether evaluating expr results in empty stderr output

so essentially I test if the expression was executed cleanly. I could drop `@test_nowarn`, but it does not harm to have it I think.

Reply:
I'd be inclined to drop these as reading them you'd think a warning could be thrown somewhere, but as you like. Could also test that the `cols` field contains what we expect it to, though it's kind of obvious.

Reply:
OK - I will change this to the test you propose.